scanner/decisions/0010-separate-versioning-for-vuln-bundles.md

```diff
@@ -0,0 +1,175 @@
+
+## Decision
+
+We will implement a versioning stream for vulnerability bundles, bumped when we determine that ClairCore's schema has changed.
+
+Initially, two version streams will exist (table 1 below):
+
+| Stream | Source Ref   | Description                                                     |
+|--------|--------------|-----------------------------------------------------------------|
+| dev    | heads/master | This stream tracks the latest vulnerability schema              |
+| v1     | tags/4.5.0   | This stream tracks the schema used by the last StackRox release |
+
+On the subsequent releases (for example, `4.6.0`), one of the two will be carried out:
+
+1.  The schema hasn't changed, so v1 now tracks the new release branch (`tags/4.6.0`).
+
+   | Stream | Source Ref   |
+   |--------|--------------|
+   | dev    | heads/master |
+   | v1     | tags/4.6.0   |
+
+2.  The schema has changed, so v1 stays at the current git ref (`tags/4.5.0`), and a new version, v2, is created to track the next release (`tags/4.6.0`).
+
+   | Stream | Source Ref   |
+   |--------|--------------|
+   | dev    | heads/master |
+   | v1     | tags/4.5.0   |
+   | v2     | tags/4.6.0   |
+
+This process could be either manual or automated.
+
+### Building versioned bundles
+
+#### Online mode
+
+The GitHub workflow will have a map with "bundle version" as keys and "source reference" as values, similar to Table 1 above.  This map can be stored statically in the matrix definition or as a file in the source repository.
+
+A source reference can be one of the following:
+
+1.  A branch reference, `heads/<branch-name>`, example: `heads/release-4.6`
+2.  A tag reference, `tags/<tag-name>`, example: `tags/4.6.0`.
+3.  A StackRox release, `release/<stackrox-release>`, example: `release/4.6.0`
+
+Both (1.) and (2.) will determine a specific commit to use to build the vulnerability exporter.
+
+(3.) will select a desired release build based on a StackRox target release.  The workflow will find the best matching Git tag for the specified `<stackrox-release>`, by checking if the version exists as a tag or matching it against release candidate (`-rc`) or patch (`.x`) tags.
+
+For each version in the mapping, a bundle will be built based on the determined git ref and stored at `https://definitions.stackrox.io/v4/<bundle-version>/vulnerabilities.zip`.
+
+In every release cut, the key specified in `scanner/VULNERABILITY_VERSION` will be updated in the mapping to the current release reference (e.g., `release/4.6.0`).  Notice that the key is still updated if the version is the same.   If it's new, it will be added, and the previous version will stay.
+
+For example, in release cut 4.6 we would look into `scanner/VULNERABILITY_VERSION`, let's say its contents are:
+
+```
+$ cat canner/VULNERABILITY_VERSION
+v1
+```
+
+The mapping should be updated so `v1` points to the current release branch reference.  Let's say the mapping was:
+
+bundle version | git reference
+--- | ---
```

Comment:
GitHub seems to have unfortunate formatting when looking at the raw `.md` file. When rendered, is this pretty?

Reply:
There's a button in the PR's diff interface you can use to check how it looks like rendered by GitHub.